### PR TITLE
Makes storage insertion failure more explicit

### PR DESCRIPTION
**Pull request by ike709**
## About the PR 

I'm really sick of `Can't insert` being the only feedback for failing to insert an item into storage. _Why_ can't I insert? This PR will tell you.

**Screenshots**

image

**Changelog**

:cl:
- tweak: Failing to insert an item into a storage container will now give you an actual failure reason, such as insufficient capacity.

### CODE REVIEW

File: Content.Server/Storage/EntitySystems/StorageSystem.cs [hunk 1]
```diff
@@ -474,29 +474,51 @@ private void RecalculateStorageUsed(ServerStorageComponent storageComp)
...
         /// <returns>true if it can be inserted, false otherwise</returns>
-        public bool CanInsert(EntityUid uid, EntityUid insertEnt, ServerStorageComponent? storageComp = null)
+        public bool CanInsert(EntityUid uid, EntityUid insertEnt, out string? reason, ServerStorageComponent? storageComp = null)
         {
             if (!Resolve(uid, ref storageComp))
+            {
+                reason = null;
                 return false;
+            }
+
```
Comment (ike709) [inline Content.Server/Storage/EntitySystems/StorageSystem.cs:486]:
```suggestion
```

File: Content.Server/Storage/EntitySystems/StorageSystem.cs [hunk 2]
```diff
@@ -474,29 +474,50 @@ private void RecalculateStorageUsed(ServerStorageComponent storageComp)
...
                 return false;
+            }
 
             if (TryComp(insertEnt, out ServerStorageComponent? storage) &&
                 storage.StorageCapacityMax >= storageComp.StorageCapacityMax)
+            {
+                reason = "comp-storage-insufficient-capacity";
                 return false;
+            }
+
```
Comment (ike709) [inline Content.Server/Storage/EntitySystems/StorageSystem.cs:493]:
```suggestion
```

**Comment by metalgearsloth** [comment]
https://github.com/space-wizards/space-station-14/pull/9291 bruh

**Comment by ike709** [comment]
Oh yeah that exists.

**Comment by EmoGarbage404** [comment]
This is fine to be merged. This and the EntityStorage pr don't overlap at all.

**Review by EmoGarbage404 — APPROVED** [review]

**Review by ElectroJr — APPROVED** [review]

**Pull request merged**